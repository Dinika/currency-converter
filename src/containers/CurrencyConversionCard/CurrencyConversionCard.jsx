import React, { Component } from 'react';
import { Segment, Button } from 'semantic-ui-react';
import CurrencyDetailForm from '../../components/CurrencyDetailForm/CurrencyDetailForm';

class CurrencyConversionCard extends Component {
    render() {
        return (
            <Segment stacked>
                <CurrencyDetailForm />
                <CurrencyDetailForm />
            </Segment>
        )
    }
}

export default CurrencyConversionCard;