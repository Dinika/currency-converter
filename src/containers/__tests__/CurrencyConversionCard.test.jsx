import React from 'react';
import { shallow } from 'enzyme';
import CurrencyConversionCard from '../CurrencyConversionCard/CurrencyConversionCard';

describe('CurrencyConversionCard', () => {
  it('renders Loader when loading is true', () => {
    const wrapper = shallow(
        <CurrencyConversionCard />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders CurrencyDetailForm when loading set to false', () => {
    const wrapper = shallow(
        <CurrencyConversionCard />
    );
    wrapper.setState({ loading: false })
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });

});
