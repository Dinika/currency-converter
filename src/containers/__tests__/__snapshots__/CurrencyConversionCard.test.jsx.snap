// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyConversionCard renders CurrencyDetailForm when loading set to false 1`] = `
<Segment
  stacked={true}
>
  <CurrencySummary
    fromCurrencyType="EUR"
    targetCurrencyAmount={0}
    targetCurrencyType="EUR"
  />
  <CurrencyDetailForm
    currencyAmount={0}
    currencyOptions={Array []}
    name="currency1"
    onCurrencyAmountChange={[Function]}
    onCurrencyTypeChange={[Function]}
  />
  <CurrencyDetailForm
    currencyAmount={0}
    currencyOptions={Array []}
    name="currency2"
    onCurrencyAmountChange={[Function]}
    onCurrencyTypeChange={[Function]}
  />
</Segment>
`;

exports[`CurrencyConversionCard renders Loader when loading is true 1`] = `<LoadingAnimation />`;
